server/chroma/src/main.rs

```diff
@@ -1,100 +1,47 @@
 extern crate core;
 
-use crate::config::Config;
-use crate::routes::appdata::{AlbumIdCache, AppData, Ratelimits, SessionIdCache, WebData};
-use crate::routes::routable::Routable;
-use actix_cors::Cors;
-use actix_web::{web, App, HttpServer};
-use cabbage::KoalaApi;
-use color_eyre::eyre::Error;
-use color_eyre::Result;
-use dal::database::Database;
-use dal::storage_engine::{S3Config, Storage};
 use dotenv::dotenv;
-use noiseless_tracing_actix_web::NoiselessRootSpanBuilder;
-use std::time::Duration;
-use tracing::{info, warn};
-use tracing_actix_web::TracingLogger;
+use tracing::level_filters::LevelFilter;
+use tracing::{error, warn};
 use tracing_subscriber::fmt::layer;
 use tracing_subscriber::layer::SubscriberExt;
 use tracing_subscriber::util::SubscriberInitExt;
 use tracing_subscriber::EnvFilter;
 
 mod config;
 mod routes;
+mod server;
 
 #[tokio::main]
-async fn main() -> Result<()> {
-    dotenv().ok();
+async fn main() {
+    // Try to load the environment variables from the .env file
+    let dotenv_err = dotenv().err();
 
-    install_tracing();
+    // Initialize the tracing logger
+    init_tracing();
 
-    info!("Starting");
-
-    info!("Parsing config");
-    let config = Config::parse()?;
-
-    if !config.validate() {
-        return Err(Error::msg("Config is not valid."));
+    // Check if the dot env was loaded correctly, otherwise send a warning
+    if let Some(err) = dotenv_err {
+        warn!("failed to load .env file: {:#}", err);
     }
 
-    if !config.service_tokens.is_empty() {
-        warn!("There are service tokens configured, Make sure these are, and stay, confidential!");
+    // Run the server
+    if let Err(err) = server::run().await {
+        error!("{}", err);
+        err.chain()
+            .skip(1)
+            .for_each(|cause| error!("because: {}", cause));
+        std::process::exit(1);
```

Comment:
Still not really a fan of `exit`, doesn't `anyhow` allow returning a `Result<T, E>` from the `main` fn as well?

Reply:
Understandably, it should be fine as there should be no more stacks to unwind when this function is called. This is what `anyhow` does under the hood when you return a `anyhow::Result<T>` from the `main` fn (https://docs.rs/anyhow/latest/anyhow/struct.Error.html). That being said, it is better to err on the safe side and rust also has support for returning an ExitCode from main which is returned at the end of `main`, so it is guaranteed that the stack will be unwound (https://doc.rust-lang.org/std/process/struct.ExitCode.html).